docs/content/manual/manual.md

```diff
@@ -0,0 +1,5538 @@
+[false, true, false, true, false]
+~~~
+
+:::
+
+### `endswith(str)` {#endswith}
+
+Outputs `true` if . ends with the given string argument.
+
+::: Examples
+
+~~~
+[.[]|endswith("foo")]
+["foobar", "barfoo"]
+[false, true]
+~~~
+
+:::
+
+### `ltrimstr(str)` {#ltrimstr}
+
+Outputs its input with the given prefix string removed, if it
+starts with it.
+
+::: Examples
+
+~~~
+[.[]|ltrimstr("foo")]
+["fo", "foo", "barfoo", "foobar", "afoo"]
+["fo","","barfoo","bar","afoo"]
+~~~
+
+:::
+
+### `rtrimstr(str)` {#rtrimstr}
+
+Outputs its input with the given suffix string removed, if it
+ends with it.
+
+::: Examples
+
+~~~
+[.[]|rtrimstr("foo")]
+["fo", "foo", "barfoo", "foobar", "foob"]
+["fo","","bar","foobar","foob"]
+~~~
+
+:::
+
+### `trim`, `ltrim`, `rtrim`
+
+`trim` trims both leading and trailing whitespace.
+
+`ltrim` trims only leading (left side) whitespace.
+
+`rtrim` trims only trailing (right side) whitespace.
+
+Whitespace characters are the usual `" "`, `"\n"` `"\t"`, `"\r"`
+and also all characters in the Unicode character database with the
+whitespace property. Note that what considers whitespace might
+change in the future.
+
+::: Examples
+
+~~~
+trim, ltrim, rtrim
+" abc "
+"abc"
+"abc "
+" abc"
+~~~
+
+:::
+
+### `explode`
+
+Converts an input string into an array of the string's codepoint numbers.
+
+::: Examples
+
+~~~
+explode
+"foobar"
+[102,111,111,98,97,114]
+~~~
+
+:::
+
+### `implode`
+
+The inverse of explode.
+
+::: Examples
+
+~~~
+implode
+[65, 66, 67]
+"ABC"
+~~~
+
+:::
+
+### `split(str)` {#split-1}
+
+Splits an input string on the separator argument.
+
+`split` can also split on regex matches when called with
+two arguments (see the regular expressions section below).
+
+::: Examples
+
+~~~
+split(", ")
+"a, b,c,d, e, "
+["a","b,c,d","e",""]
+~~~
+
+:::
+
+### `join(str)` {#join}
+
+Joins the array of elements given as input, using the
+argument as separator. It is the inverse of `split`: that is,
+running `split("foo") | join("foo")` over any input string
+returns said input string.
+
+Numbers and booleans in the input are converted to strings.
+Null values are treated as empty strings. Arrays and objects
+in the input are not supported.
+
+::: Compatibility
+When given an array `[x0, x1, ..., xn]`,
+in jq, `join(x)` converts all elements of the input array to strings and intersperses them with `x`, whereas
+in jaq, `join(x)` simply calculates `x0 + x + x1 + x + ... + xn`.
+When all elements of the input array and `x` are strings, jq and jaq yield the same output.
+:::
+
+::: Examples
+
+~~~
+join(", ")
+["a","b,c,d","e"]
+"a, b,c,d, e"
+~~~
+
+~~~
+join(" ")
+["a",1,2.3,true,null,false]
+"a 1 2.3 true  false"
+~~~
+
+:::
+
+### `ascii_downcase`, `ascii_upcase`
+
+Emit a copy of the input string with its alphabetic characters (a-z and A-Z)
+converted to the specified case.
+
+::: Examples
+
+~~~
+ascii_upcase
+"useful but not for é"
+"USEFUL BUT NOT FOR é"
+~~~
+
+:::
+
+## String formatting functions
```

Comment:
I think these are called string formats. They can be used like filters, but when placed before strings they can format interpolations, so different concept from functions.

Reply:
The jq 1.7 manual says: "The @foo syntax is used to format and escape strings [...]", but it never really gives an explicit name for `@...`. You are right that they are not completely like functions, because they can be used before strings (unlike regular functions), but they can be *also* used like functions. I added a note about this to the manual now, but left the name of the section unchanged. Please let me know if you are happy with that, otherwise I can still change it.